lib/clientv2/client.go

```diff
@@ -153,12 +165,134 @@ func (p poolAccessor) GetEnabledPools(ipVersion int) ([]net.IPNet, error) {
 // Most Calico deployment scenarios will automatically implicitly invoke this
 // method and so a general consumer of this API can assume that the datastore
 // is already initialized.
-func (c client) EnsureInitialized() error {
+func (c client) EnsureInitialized(ctx context.Context, calicoVersion, clusterType string) error {
 	// Perform datastore specific initialization first.
 	if err := c.backend.EnsureInitialized(); err != nil {
 		return err
 	}
 
+	if err := c.ensureClusterInformation(ctx, calicoVersion, clusterType); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+const globalClusterInfoName = "default"
+
+// ensureClusterInformation ensures that the ClusterInformation fields i.e. ClusterType,
```

Comment:
Given that we have this now, should we be calling this from the Node create method to reinstate the HostEndpoint use case.